src/main/java/com/patchworkmc/event/EventHandlerRewriter.java

```diff
@@ -68,34 +101,79 @@ public MethodVisitor visitMethod(int access, String name, String descriptor, Str
 
 	@Override
 	public void visitEnd() {
-		// Write patchwork_registerStaticEventHandlers
+		genStaticRegistrar();
+		genInstanceRegistrar();
+		super.visitEnd();
+		finished = true;
+	}
+
+	private void genStaticRegistrar() {
+		if (staticSubscribeEvents.isEmpty()) {
+			return;
+		}
+
+		MethodVisitor staticRegistrar = super.visitMethod(Opcodes.ACC_PUBLIC | Opcodes.ACC_STATIC, EventConstants.REGISTER_STATIC, EventConstants.REGISTER_STATIC_DESC, null, null);
```

Comment:
super.visitMethod can return `null` if it is not interested in the generated method, this should be supported with a null check. Same for genInstanceRegistrar.